test/e2e/bandwidth_test.go

```diff
@@ -16,41 +16,82 @@ package e2e
 
 import (
 	"fmt"
-	"strconv"
 	"strings"
 	"testing"
+
+	v1 "k8s.io/api/core/v1"
 )
 
-// BenchmarkBandwidthIntraNode runs the benchmark of bandwidth between Pods on same node.
-func BenchmarkBandwidthIntraNode(b *testing.B) {
-	withPerfTestSetup(func(data *TestData) {
-		b.StartTimer()
-
-		podDefA := createPerfTestPodDefinition("perftest-a", perftoolContainerName, perftoolImage)
-		if _, err := data.clientset.CoreV1().Pods(testNamespace).Create(podDefA); err != nil {
-			b.Fatalf("Error when creating the first perftest Pod: %v", err)
-		}
-		podDefB := createPerfTestPodDefinition("perftest-b", perftoolContainerName, perftoolImage)
-		if _, err := data.clientset.CoreV1().Pods(testNamespace).Create(podDefB); err != nil {
-			b.Fatalf("Error when creating the second perftest Pod: %v", err)
-		}
-		podBIP, err := data.podWaitForIP(defaultTimeout, podDefB.Name, testNamespace)
-		if err != nil {
-			b.Fatalf("Error when getting perftest Pod IP: %v", err)
-		}
-		stdout, _, err := data.runCommandFromPod(testNamespace, "perftest-a", perftoolContainerName, []string{"bash", "-c", fmt.Sprintf("iperf3 -c %s|grep sender|awk '{print $7,$8}'", podBIP)})
-		if err != nil {
-			b.Fatalf("Error when running iperf3 client: %v", err)
-		}
-		stdout = strings.TrimSpace(stdout)
-		results := strings.Split(stdout, " ")
-		if len(results) != 2 {
-			b.Fatalf("Error when parsing iperf result: cannot parse output `%s`", stdout)
-		}
-		// Disable default output.
-		b.ReportMetric(0, "ns/op")
-		bandwidthNum, _ := strconv.ParseFloat(results[0], 64)
-		bandwidthUnit := strings.TrimSpace(results[1])
-		b.ReportMetric(bandwidthNum, bandwidthUnit)
-	}, b)
+const iperfPort = 5201
+
+// TestBenchmarkBandwidthIntraNode runs the benchmark of bandwidth between Pods on same node.
+func TestBenchmarkBandwidthIntraNode(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	if err := data.createPodOnNode("perftest-a", masterNodeName(), perftoolImage, nil, nil, nil, nil); err != nil {
+		t.Fatalf("Error when creating the perftest client Pod: %v", err)
+	}
+	if err := data.podWaitForRunning(defaultTimeout, "perftest-a", testNamespace); err != nil {
+		t.Fatalf("Error when waiting for the perftest client Pod: %v", err)
+	}
+	if err := data.createPodOnNode("perftest-b", masterNodeName(), perftoolImage, nil, nil, nil, []v1.ContainerPort{{Protocol: v1.ProtocolTCP, ContainerPort: iperfPort}}); err != nil {
+		t.Fatalf("Error when creating the perftest server Pod: %v", err)
+	}
+	podBIP, err := data.podWaitForIP(defaultTimeout, "perftest-b", testNamespace)
+	if err != nil {
+		t.Fatalf("Error when getting the perftest server Pod IP: %v", err)
```

Comment:
s/Pod IP/Pod's IP

here and everywhere in the file

Reply:
Done.